### ISSUE TITLE
Value cannot be null. Parameter name: securityNamespace 

### ISSUE THREAD

**Issue by drphrozen** [microsoft/azure-pipelines-tasks#11291]
## Note  
Issues in this repo are for tracking bugs, feature requests and questions for the tasks in this repo 

For a list:  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

If you have an issue or request for the Azure Pipelines service, use developer community instead:  

https://developercommunity.visualstudio.com/spaces/21/index.html )

## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: CacheBeta

list here (V# not needed):  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

## Environment
- Server - Azure Pipelines or TFS on-premises?
    
    - If using Azure Pipelines, provide the account name, team project name, build definition name/build number: 
https://dev.azure.com/kvadrat-project/Kvadrat/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=3
https://dev.azure.com/kvadrat-project/Kvadrat/_build/results?buildId=3292

- Agent - Hosted or Private: 
    
    - If using Hosted agent, provide agent queue name: Hosted VS2017

    - If using private agent, provide the OS of the machine running the agent and the agent version: 

## Issue Description

[Include task name(s), screenshots and any other relevant details]

![image](https://user-images.githubusercontent.com/309670/64325016-b09c8500-cfc7-11e9-9466-e005e4cfcab5.png)

![image](https://user-images.githubusercontent.com/309670/64324741-33711000-cfc7-11e9-9b78-ac0e4dc68e32.png)

### Task logs

```
##[Error 1]
Value cannot be null.
Parameter name: securityNamespace 
```
[Enable debug logging and please provide the zip file containing all the logs for a speedy resolution]

## Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: https://docs.microsoft.com/en-us/vsts/build-release/actions/troubleshooting

### Error logs

[Insert error from the logs here for a quick overview]

**Comment by evanjd** [comment]
I was able to resolve this by adding the following property to the CacheBeta task.

```yaml
securityNamespace: cache
```

Definitely seems like a bug though - this prop is not visible in the form and triggers validation errors.

**Comment by johnterickson** [comment]
Thanks for reporting this @drphrozen and thanks for the workaround @evanjd !

@fadnavistanmay is this on our radar?

**Comment by fadnavistanmay** [comment]
Hi @drphrozen - is still still an issue. IIRC we had released a fix for this.

**Comment by fadnavistanmay** [comment]
I am closing this issue. Please feel free to re-open if it still persists.

Thanks.

**Comment by pereiren** [comment]
Hi,

I was having this issue when adding another GitHub organization to my Azure DevOps pipelines.
![image](https://user-images.githubusercontent.com/11976485/70394114-08e18980-19f2-11ea-90c1-b838b77d9a59.png)

![image](https://user-images.githubusercontent.com/11976485/70394119-172fa580-19f2-11ea-9ea4-8834a059515e.png)


I grabbed the adquisitionID from the URL and pasted here and it worked
![image](https://user-images.githubusercontent.com/11976485/70394149-5eb63180-19f2-11ea-916f-35d22a502ac2.png)

Hope it helps, I cannot reproduce it now but if I face it again I will provide more data

**Comment by msebolt** [comment]
We're still seeing this issue, see https://github.com/MicrosoftDocs/azure-devops-docs/issues/8146.

**Comment by hansmbakker** [comment]
@fadnavistanmay can you please reopen this? The issue still exists

#please-reopen

**Comment by ibedritskiy** [comment]
@hansmbakker I have the same issue. Could you please guide me where to add the `securityNamespace: cache`?

I added it here so it didn't work:
`  - task: Cache@2
    displayName: 'NuGet Cache'
    inputs:
      securityNamespace: cache
      key: 'nuget | "$(Agent.OS)" | **/packages.lock.json,!**/bin/**,!**/obj/**'
      restoreKeys: |
        nuget | "$(Agent.OS)"
        nuget
      path: '$(NUGET_PACKAGES)'
      cacheHitVar: 'CACHE_RESTORED'`
      
 Ended up here by following the guide:
 https://learn.microsoft.com/en-us/azure/devops/pipelines/artifacts/caching-nuget?view=azure-devops

**Comment by chrisbarton126** [comment]
I've experienced this same issue when porting existing pipelines between two organisations in Azure DevOps. In the original organisation the issue with `Cache@2` did not exist, whereas in the new organisation, the exact same pipeline is reporting an error regarding `Value cannot be null. Parameter name: securityNamespace`.

Does anybody know why this would happen between two separate organisations? Could it perhaps be related to security/permissions at org level? Like @ibedritskiy, I attempted to add `securityNamespace: cache` within the `inputs` block, but this had no effect and still failed.
